src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2390,6 +2390,22 @@ private function registerNotifierConfiguration(array $config, ContainerBuilder $
             $container->removeDefinition('notifier.channel.email');
         }
 
+        $servicesWithBusArgument = ['texter', 'chatter', 'notifier.channel.chat', 'notifier.channel.email', 'notifier.channel.sms'];
+        if (false === $messageBus = $config['message_bus']) {
+            foreach ($servicesWithBusArgument as $serviceId) {
+                if ($container->hasDefinition($serviceId)) {
+                    $container->getDefinition($serviceId)->replaceArgument(1, null);
+                }
+            }
+        } else {
+            foreach ($servicesWithBusArgument as $serviceId) {
+                if ($container->hasDefinition($serviceId)) {
+                    $container->getDefinition($serviceId)->setArgument(0, null);
+                    $container->getDefinition($serviceId)->replaceArgument(1, $messageBus ? new Reference($messageBus) : new Reference('messenger.default_bus', ContainerInterface::NULL_ON_INVALID_REFERENCE));
```

Comment:
```suggestion
                    $container->getDefinition($serviceId)
                        ->setArgument(0, null);
                        ->replaceArgument(1, $messageBus ? new Reference($messageBus) : new Reference('messenger.default_bus', ContainerInterface::NULL_ON_INVALID_REFERENCE))
                    ;
```

?

Reply:
```diff
- ->setArgument(0, null);
+ ->setArgument(0, null)
```
?

Reply:
@OskarStark : Right, fixed the suggestion

Reply:
Thanks @ogizanagi and @OskarStark